client/core/core.go

```diff
@@ -2698,18 +2702,28 @@ func (dc *dexConnection) bondOpts() (assetID uint32, targetTier, max uint64) {
 	return dc.acct.bondAsset, dc.acct.targetTier, dc.acct.maxBondedAmt
 }
 
-func (dc *dexConnection) hasActiveBond(assetID uint32) bool {
-	check := func(bonds []*db.Bond) bool {
+func (dc *dexConnection) bondTotalInternal(assetID uint32) (total, active uint64) {
+	sum := func(bonds []*db.Bond) (amt uint64) {
 		for _, b := range bonds {
 			if assetID == b.AssetID {
-				return true
+				amt += b.Amount
 			}
 		}
-		return false
+		return
 	}
+	active = sum(dc.acct.bonds)
+	return active + sum(dc.acct.pendingBonds) + sum(dc.acct.expiredBonds), active
+}
+
+func (dc *dexConnection) bondTotal(assetID uint32) (total, active uint64) {
 	dc.acct.authMtx.RLock()
 	defer dc.acct.authMtx.RUnlock()
-	return check(dc.acct.bonds) || check(dc.acct.pendingBonds) || check(dc.acct.expiredBonds)
+	return dc.bondTotalInternal(assetID)
+}
+
+func (dc *dexConnection) hasActiveBond(assetID uint32) bool {
+	total, _ := dc.bondTotal(assetID)
```

Comment:
This should check `active > 0`, not `total > 0`.

Reply:
Hmm.  I guess I need two methods, or to rename this one to `hasUnspentBond`, because in the one current caller (`isActiveBondAsset`, just below), it's in the branch where: `includeLive should be set to consider all existing unspent bonds that need to be refunded in the future (only requires a broadcast, no wallet signing ability).` where the idea is you don't want to disable the wallet if you even need to refund an expired bond.

Reply:
Yeah, then rename would suffice.